---
title: "Diet and the Gut Microbiome in Patients with Parkinson’s Disease"
author: "Dayoon Kwon"
date: "2023-12-22"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(vegan)
library(phyloseq)
library(ggplot2)
library(Maaslin2)
library(microbiome)
library(dada2)
library(Biostrings)
library(ggpubr)
library(png)
```

# Introduction

This file documents code used to perform the main statistical analyses detailed in the manuscript ***Diet and the Gut Microbiome in Patients with Parkinson’s Disease***.

# Bioinformatic processing of sequences

Sequence reads were quality-filtered using the [DADA2 pipeline](https://benjjneb.github.io/dada2/tutorial.html)

```{r}
# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
fnFs

# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
sample.names

plotQualityProfile(fnFs[1:2])

plotQualityProfile(fnRs[1:2])

# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,140),trimLeft=c(10,10), 
                     maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE, compress=TRUE, multithread=FALSE) # On Windows set multithread=FALSE
head(out)

errF <- learnErrors(filtFs, multithread=TRUE)
errR <- learnErrors(filtRs, multithread=TRUE)

plotErrors(errF, nominalQ=TRUE)
plotErrors(errR, nominalQ=TRUE)

dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
dadaRs <- dada(filtRs, err=errR, multithread=TRUE)

dadaFs[[1]]

mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)

# Inspect the merger data.frame from the first sample
head(mergers[[1]])

seqtab <- makeSequenceTable(mergers)
dim(seqtab)

# Inspect distribution of sequence lengths
table(nchar(getSequences(seqtab)))

seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
dim(seqtab.nochim)
sum(seqtab.nochim)/sum(seqtab)

getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))

# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
head(track)

taxa <- assignTaxonomy(seqtab.nochim, "silva_nr_v132_train_set.fa.gz", multithread=TRUE)
taxa <- addSpecies(taxa, "silva_species_assignment_v132.fa.gz")

taxa.print <- taxa # Removing sequence rownames for display only
rownames(taxa.print) <- NULL
head(taxa.print)

theme_set(theme_bw())

samdf <- read.csv("Meta219_11132021_full.csv", row.names=1)

ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), 
               sample_data(samdf), 
               tax_table(taxa))

ps <- prune_samples(sample_names(ps) != "Mock", ps) # Remove mock sample

dna <- Biostrings::DNAStringSet(taxa_names(ps))
names(dna) <- taxa_names(ps)
ps <- merge_phyloseq(ps, dna)
taxa_names(ps) <- paste0("ASV", seq(ntaxa(ps)))
ps
```

# Setting up R environment for statistical analyses

## Create needed functions for analyses

```{r}
#prune_by_group function
prune_by_group <- function(phy, group, P = 0.5){
  
  grs <- sample_data(phy)[, group][[1]]
  grsu <- unique(grs)
  
  these <- list()
  for(i in seq_along(grsu)){
    subotu <- t(otu_table(phy,taxa_are_rows=FALSE))[, grs == grsu[i]]
    subcount <- apply(subotu, 1, function(x) sum(x > 0))
    these[[i]] <- names(subcount[subcount >= ceiling(sum(grs == grsu[i])*P)])
  }
  
  this <- unique(do.call(c, these))
  phynew <- prune_taxa(this, phy)
  return(phynew)
}

#alpha diversity function
alpha_func=function(x,y,z) {
  c(paste(round(x[2,1],2), " (",round(x[2,1]-1.96*x[2,2],2),", ",round(x[2,1]+1.96*x[2,2],2),")",sep=""),
    paste(round(x[3,1],2), " (",round(x[3,1]-1.96*x[3,2],2),", ",round(x[3,1]+1.96*x[3,2],2),")",sep=""),
    round(y[2,4],3),
    paste(round(z[2,1],2), " (",round(z[2,1]-1.96*z[2,2],2),", ",round(z[2,1]+1.96*z[2,2],2),")",sep=""),
    round(z[2,4],3))
}

#differential abundance function
MaAsLin2.plus = function(ps, output, 
                          fixed_effects=NULL, 
                          reference=NULL){
  
  input_data = data.frame(otu_table(ps))
  input_metadata = data.frame(sample_data(ps))
  taxadata = data.frame(tax_table(ps))
  taxadata$feature = rownames(taxadata)
  exposure = fixed_effects[5]
  
  fits = Maaslin2(input_data,
                   input_metadata,
                   output=output,
                   min_abundance=0.01,
                   min_prevalence=0,
                   normalization='TMM',
                   transform='NONE',
                   analysis_method='NEGBIN',
                   fixed_effects=fixed_effects,
                   standardize=FALSE,
                   plot_heatmap=FALSE,
                   plot_scatter=FALSE,
                   heatmap_first_n=50,
                   reference=reference)
  
  res = fits$results %>%
    filter(metadata==exposure) %>%
    merge(., taxadata, by="feature") %>%
    select(-feature,-metadata,-value,-name,-N,-N.not.zero) %>%
    mutate(qval=p.adjust(as.numeric(as.character(pval)), method="fdr"),
           fc = coef/log(2),
           fc_ll = (coef-(1.96*stderr))/log(2),
           fc_up = (coef+(1.96*stderr))/log(2)) 

  return(res)
}
```

# Filtering

```{r}
any(taxa_sums(ps) == 0)
ps1 <- prune_taxa(taxa_sums(ps) > 0, ps)
ntaxa(ps)
ntaxa(ps1)

#Rarefy for Alpha diversirt and Presence/Absence beta diversity
rarecurve(t(otu_table(ps)), step=50, cex=0.5)
ps.rarefied = rarefy_even_depth(ps, rngseed=1, sample.size=0.9*min(sample_sums(ps)), replace=F)

#Whole phylum filtering
#Remove of the feature with ambiguous phylum annotation.
ps2 <- subset_taxa(ps1, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))
ps2

#Generate a prevalence table (number of samples each taxa occurs in) for each taxa.
prevdf = apply(X = otu_table(ps2),
               MARGIN = 2,
               FUN = function(x){sum(x > 0)})

prevdf = data.frame(Prevalence = prevdf,
                    TotalAbundance = taxa_sums(ps2),
                    tax_table(ps2))

#Investigate low prevalence/abundance phylum and subset them out.
phylum_prev<-plyr::ddply(prevdf, "Phylum", function(df1){
  data.frame(mean_prevalence=mean(df1$Prevalence),total_abundance=sum(df1$TotalAbundance,na.rm = T),stringsAsFactors = F)
})
phylum_prev<-phylum_prev[order(-phylum_prev$total_abundance),]
phylum_prev

#Using the table above, determine the genera to filter
#Low abundance phyla (<50)
phyla2Filter = subset(phylum_prev,total_abundance<50)$Phylum

ps3 = subset_taxa(ps2, !Phylum %in% phyla2Filter)
ps3


#Individual Taxa Filtering
#Subset to the remaining phyla by prevalence.
prevdf1 = subset(prevdf, Phylum %in% get_taxa_unique(ps3, taxonomic.rank = "Phylum"))
p_revdf <- ggplot(prevdf1, aes(TotalAbundance, Prevalence / nsamples(ps3),color=Phylum)) +
  # Include a guess for parameter
  geom_hline(yintercept = 0.05, alpha = 0.5, linetype = 2) + geom_point(size = 2, alpha = 0.7) +
  scale_x_log10() +  xlab("Total Abundance") + ylab("Prevalence [Frac. Samples]") +
  facet_wrap(~Phylum) + theme(legend.position="none")


# Define prevalence threshold as 20% of total samples
prevalenceThreshold = 0.20 * nsamples(ps3)
prevalenceThreshold
# Execute prevalence filter, using `prune_taxa()` function
keepTaxa = rownames(prevdf1)[(prevdf1$Prevalence >= prevalenceThreshold)]
ps4 = prune_taxa(keepTaxa, ps3)
ps4
summary(rowSums(otu_table(ps4)))
sum(rowSums(otu_table(ps4)))
sd(rowSums(otu_table(ps4)))
```

# Statistical analyses

## Alpha diversity analysis

```{r}
shannon <- vegan::diversity(ps4@otu_table, index = "shannon")
alphadiv <- cbind(ps4@sam_data, shannon)

x1=summary(lm(shannon~hei_cat+age+sex+race+platform,data=alphadiv))$coefficients
x1.1=summary(lm(shannon~hei_cat2+age+sex+race+platform,data=alphadiv))$coefficients
x1.2=summary(lm(shannon~hei_iqr+age+sex+race+platform,data=alphadiv))$coefficients

x2=summary(lm(shannon~fiber_cat+age+sex+race+platform,data=alphadiv))$coefficients
x2.1=summary(lm(shannon~fiber_cat2+age+sex+race+platform,data=alphadiv))$coefficients
x2.2=summary(lm(shannon~fiber+age+sex+race+platform,data=alphadiv))$coefficients

x3=summary(lm(shannon~sugar_cat+age+sex+race+platform,data=alphadiv))$coefficients
x3.1=summary(lm(shannon~sugar_cat2+age+sex+race+platform,data=alphadiv))$coefficients
x3.2=summary(lm(shannon~sugar+age+sex+race+platform,data=alphadiv))$coefficients

table1=rbind(alpha_func(x1,x1.1,x1.2),alpha_func(x2,x2.1,x2.2),alpha_func(x3,x3.1,x3.2))

hei_alpha = alphadiv %>%
  ggplot(.,aes(x = hei_cat, y = shannon, fill = hei_cat)) +
  geom_boxplot(outlier.color = NULL,lwd=1) +
  scale_fill_manual(name = "HEI Tertile", values = c("#EF6548","#FDBF6F","#41B6C4"), labels=c("Low","Intermediate","High")) +
  scale_x_discrete(breaks=c("0","1","2"),labels=c("Low", "Intermediate", "High")) +
  labs(x = "HEI Tertile", y = "Shannon Index") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,color="black"),axis.text.y=element_text(size=18,color="black"),
        legend.text = element_text(size=18,color="black"),legend.title = element_text(size=18, face="bold")) +
  guides(fill="none") +
  annotate("text", x = 3.2, y = 1.8, label = "P-trend = 0.603", size=5)

fiber_alpha = ggplot(alphadiv,aes(x = fiber_cat, y = shannon, fill = fiber_cat)) +
  geom_boxplot(outlier.color = NULL,lwd=1) +
  scale_fill_manual(name = "Dietary Fiber Tertile", values = c("#EF6548","#FDBF6F","#41B6C4"), labels=c("Low","Intermediate","High")) +
  scale_x_discrete(breaks=c("0","1","2"),labels=c("Low", "Intermediate", "High")) +
  labs(x = "Dietary Fiber Tertile", y = "Shannon Index") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,color="black"),axis.text.y=element_text(size=18,color="black"),
        legend.text = element_text(size=18,color="black"),legend.title = element_text(size=18, face="bold")) +
  guides(fill="none")+
  annotate("text", x = 3.2, y = 1.8, label = "P-trend = 0.888", size=5)

sugar_alpha = ggplot(alphadiv,aes(x = sugar_cat, y = shannon, fill = sugar_cat)) +
  geom_boxplot(outlier.color = NULL,lwd=1) +
  scale_fill_manual(name = "Added Sugar Tertile", values = c("#EF6548","#FDBF6F","#41B6C4"), labels=c("Low","Intermediate","High")) +
  scale_x_discrete(breaks=c("0","1","2"),labels=c("Low", "Intermediate", "High")) +
  labs(x = "Added Sugar Tertile", y = "Shannon Index") +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,color="black"),axis.text.y=element_text(size=18,color="black"),
        legend.text = element_text(size=18,color="black"),legend.title = element_text(size=18, face="bold")) +
  guides(fill="none") +
  annotate("text", x = 3.2, y = 1.8, label = "P-trend = 0.364", size=5)
```

## Principal Component Analysis

```{r, warning=FALSE, message=FALSE}
data_otu_filt_rar = data.frame(otu_table(ps4)) # create a separated file
dist_bc = as.matrix(vegdist(data_otu_filt_rar, method = "bray")) 

# a peek at the first five rows / columns
dist_bc[1:5, 1:5]

# calculate PCOA using Phyloseq package
pcoa_bc = ordinate(ps4, "PCoA", "bray") 
mds_dat<-data.frame(pcoa_bc$vectors)
mds_dat$PEGID=row.names(mds_dat)
mds2<-merge(mds_dat,data.frame(ps4@sam_data),by="PEGID")

mody <- capscale(pcoa_bc$vectors~ 1)
eig=eigenvals(mody)
head(eig / sum(eig))

clr1 =head(eig / sum(eig))[1]
clr2 =head(eig / sum(eig))[2]

set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+hei_cat2,
       data=d, permutations=9999, method="bray")$aov.tab
set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+fiber_cat2,
       data=d, permutations=9999, method="bray")$aov.tab
set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+sugar_cat2,
       data=d, permutations=9999, method="bray")$aov.tab

set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+hei_iqr,
       data=d, permutations=9999, method="bray")$aov.tab
set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+fiber,
       data=d, permutations=9999, method="bray")$aov.tab
set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+sugar,
       data=d, permutations=9999, method="bray")$aov.tab

hei_beta = plot_ordination(ps4, pcoa_bc, type="samples", color="hei_cat", shape = "hei_cat") + 
  geom_point(size = 4.5,) +
  stat_ellipse(aes(group = hei_cat), linetype = 2, size=1) +
  geom_vline(xintercept=0, linetype="dotted", alpha = 0.8) +
  geom_hline(yintercept=0, linetype="dotted", alpha = 0.8) +
  labs(x ="PC-1 (8.7%)", y = "PC-2 (7.3%)") +
  xlim(c(-0.5,0.5)) + 
  ylim(c(-0.5,0.5)) +
  geom_point(aes(mean(mds2$Axis.1[mds2$hei_cat==0]),mean(mds2$Axis.2[mds2$hei_cat==0])), color="black", fill="#EF6548", size=5.5, pch=21) +
  geom_point(aes(mean(mds2$Axis.1[mds2$hei_cat==1]),mean(mds2$Axis.2[mds2$hei_cat==1])), color="black", fill="#FDBF6F", size=5.5, pch=24) +
  geom_point(aes(mean(mds2$Axis.1[mds2$hei_cat==2]),mean(mds2$Axis.2[mds2$hei_cat==2])), color="black", fill="#41B6C4", size=5.5, pch=22) +
  scale_color_manual(name = "HEI Tertile", values = c("#EF6548","#FDBF6F","#41B6C4"), labels=c("Low","Intermediate","High")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),
        panel.background = element_blank(),legend.text = element_text(size=18,color="black"),
        legend.title = element_text(size=18, face="bold"),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,color="black"),axis.text.y=element_text(size=18,color="black")) +
  guides(shape = "none")+
  annotate("text", x = 0.3, y = -0.46, label = "R2 = 0.017\nP-trend = 0.025", size=5, hjust = 0)

fiber_beta = plot_ordination(ps4, pcoa_bc, type="samples", color="fiber_cat", shape = "fiber_cat") + 
  geom_point(size = 4.5,) +
  stat_ellipse(aes(group = fiber_cat), linetype = 2, size=1) +
  geom_vline(xintercept=0, linetype="dotted", alpha = 0.8) +
  geom_hline(yintercept=0, linetype="dotted", alpha = 0.8) +
  labs(x ="PC-1 (8.7%)", y = "PC-2 (7.3%)") +
  xlim(c(-0.5,0.5)) + 
  ylim(c(-0.5,0.5)) +
  geom_point(aes(mean(mds2$Axis.1[mds2$fiber_cat==0]),mean(mds2$Axis.2[mds2$fiber_cat==0])), color="black", fill="#EF6548", size=5.5, pch=21) +
  geom_point(aes(mean(mds2$Axis.1[mds2$fiber_cat==1]),mean(mds2$Axis.2[mds2$fiber_cat==1])), color="black", fill="#FDBF6F", size=5.5, pch=24) +
  geom_point(aes(mean(mds2$Axis.1[mds2$fiber_cat==2]),mean(mds2$Axis.2[mds2$fiber_cat==2])), color="black", fill="#41B6C4", size=5.5, pch=22) +
  scale_color_manual(name = "Dietary Fiber Tertile", values = c("#EF6548","#FDBF6F","#41B6C4"), labels=c("Low","Intermediate","High")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),
        panel.background = element_blank(),legend.text = element_text(size=18,color="black"),
        legend.title = element_text(size=18, face="bold"),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,color="black"),axis.text.y=element_text(size=18,color="black")) +
  guides(shape = "none") +
  annotate("text", x = 0.3, y = -0.46, label = "R2 = 0.019\nP-trend = 0.011", size=5, hjust = 0)

sugar_beta = plot_ordination(ps4, pcoa_bc, type="samples", color="sugar_cat", shape = "sugar_cat") + 
  geom_point(size = 4.5,) +
  stat_ellipse(aes(group = sugar_cat), linetype = 2, size=1) +
  geom_vline(xintercept=0, linetype="dotted", alpha = 0.8) +
  geom_hline(yintercept=0, linetype="dotted", alpha = 0.8) +
  labs(x ="PC-1 (8.7%)", y = "PC-2 (7.3%)") +
  xlim(c(-0.5,0.5)) + 
  ylim(c(-0.5,0.5)) +
  geom_point(aes(mean(mds2$Axis.1[mds2$sugar_cat==0]),mean(mds2$Axis.2[mds2$sugar_cat==0])), color="black", fill="#EF6548", size=5.5, pch=21) +
  geom_point(aes(mean(mds2$Axis.1[mds2$sugar_cat==1]),mean(mds2$Axis.2[mds2$sugar_cat==1])), color="black", fill="#FDBF6F", size=5.5, pch=24) +
  geom_point(aes(mean(mds2$Axis.1[mds2$sugar_cat==2]),mean(mds2$Axis.2[mds2$sugar_cat==2])), color="black", fill="#41B6C4", size=5.5, pch=22) +
  scale_color_manual(name = "Added Sugar Tertile", values = c("#EF6548","#FDBF6F","#41B6C4"), labels=c("Low","Intermediate","High")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),
        panel.background = element_blank(),legend.text = element_text(size=18,color="black"),
        legend.title = element_text(size=18, face="bold"),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,color="black"),axis.text.y=element_text(size=18,color="black")) +
  guides(shape = "none") +
  annotate("text", x = 0.3, y = -0.46, label = "R2 = 0.017\nP-trend = 0.044", size=5, hjust = 0)
```

## Relative abundance analysis

### Phylum level relative abundance

```{r, warning=FALSE, message=FALSE}
mycolors=c("#A6CEE3","#1F78B4","#B2DF8A","#33A02C","#FB9A99","#E31A1C",
           "#FDBF6F","#FF7F00","#CAB2D6","#6A3D9A","#FFFF99","#E6AB02",
           "#E5C494","#B15928","#D9D9D9","#B3B3B3","#80B1D3","#386CB0",
           "#8DD3C7","#1B9E77","#FCCDE5","#FB8072","#BEBADA","#BC80BD",
           "#FFD92F","#08306B")

physeqPhylum = tax_glom(ps4, "Phylum")
physeqPhylumRA = transform_sample_counts(physeqPhylum, function(x) x/sum(x))
physeqPhylumRAF = filter_taxa(physeqPhylumRA, function(x) mean(x) > 0.01, TRUE)
keepPhyla = get_taxa_unique(physeqPhylumRAF, "Phylum")
physeqF = subset_taxa(ps4, Phylum  %in% keepPhyla) 

hei_phylum = physeqF %>%
  aggregate_taxa(level = "Phylum") %>%  
  microViz::tax_sort(by="name") %>%
  microbiome::transform(transform = "compositional") %>%
  plot_composition(.,average_by = "hei_cat") + 
  guides(fill = guide_legend(ncol = 1)) + 
  labs(x = "HEI Tertile", y = "Relative abundance") +
  scale_fill_brewer("Phylum", palette = "Paired") +
  scale_x_discrete(breaks=c("0","1","2"),labels=c("Low", "Intermediate", "High")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,angle = -45, hjust=0),axis.text.y=element_text(size=18,color="black"),
        legend.title = element_text(size=18,face="bold"), legend.text = element_text(size=18))

fiber_phylum = physeqF %>%
  aggregate_taxa(level = "Phylum") %>%  
  microViz::tax_sort(by="name") %>%
  microbiome::transform(transform = "compositional") %>%
  plot_composition(.,average_by = "fiber_cat") + 
  guides(fill = guide_legend(ncol = 1)) + 
  labs(x = "Dietary Fiber Tertile", y = "Relative abundance") +
  scale_fill_brewer("Phylum", palette = "Paired") +
  scale_x_discrete(breaks=c("0","1","2"),labels=c("Low", "Intermediate", "High")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,angle = -45, hjust=0),axis.text.y=element_text(size=18,color="black"),
        legend.title = element_text(size=18,face="bold"), legend.text = element_text(size=18)) 

sugar_phylum = physeqF %>%
  aggregate_taxa(level = "Phylum") %>% 
  microViz::tax_sort(by="name") %>%
  microbiome::transform(transform = "compositional") %>%
  plot_composition(.,average_by = "sugar_cat") + 
  guides(fill = guide_legend(ncol = 1)) + 
  labs(x = "Added Sugar Tertile", y = "Relative abundance") +
  scale_fill_brewer("Phylum", palette = "Paired") +
  scale_x_discrete(breaks=c("0","1","2"),labels=c("Low", "Intermediate", "High")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,angle = -45, hjust=0),axis.text.y=element_text(size=18,color="black"),
        legend.title = element_text(size=18,face="bold"), legend.text = element_text(size=18)) 
```

### Genus level relative abundance

```{r, warning=FALSE, message=FALSE}
physeqGenus = tax_glom(ps4, "Genus")
physeqGenusRA = transform_sample_counts(physeqGenus, function(x) x/sum(x))
physeqGenusRAF = filter_taxa(physeqGenusRA, function(x) mean(x) > 0.01, TRUE)
keepGenus = get_taxa_unique(physeqGenusRAF, "Genus")
physeqG = subset_taxa(ps4, Genus  %in% keepGenus) 
physeqG@tax_table = sub("_", " ", physeqG@tax_table)

hei_genus = physeqG %>%
  aggregate_taxa(level = "Genus") %>%  
  microViz::tax_sort(by="name") %>%
  microbiome::transform(transform = "compositional") %>%
  plot_composition(.,average_by = "hei_cat") + 
  guides(fill = guide_legend(ncol = 1)) + 
  labs(x = "HEI Tertile", y = "Relative abundance") +
  scale_fill_manual("Genus", values = mycolors) +
  scale_x_discrete(breaks=c("0","1","2"),labels=c("Low", "Intermediate", "High")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,angle = -45, hjust=0),axis.text.y=element_text(size=18,color="black"),
        legend.title = element_text(size=18,color="black",face="bold"), legend.text = element_text(size=18,color="black")) 

fiber_genus = physeqG %>%
  aggregate_taxa(level = "Genus") %>%  
  microViz::tax_sort(by="name") %>%
  microbiome::transform(transform = "compositional") %>%
  plot_composition(.,average_by = "fiber_cat") + 
  guides(fill = guide_legend(ncol = 1)) + 
  labs(x = "Dietary Fiber Tertile", y = "Relative abundance") +
  scale_fill_manual("Genus", values = mycolors) +
  scale_x_discrete(breaks=c("0","1","2"),labels=c("Low", "Intermediate", "High")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,angle = -45, hjust=0),axis.text.y=element_text(size=18,color="black"),
        legend.title = element_text(size=18,color="black",face="bold"), legend.text = element_text(size=18,color="black")) 

sugar_genus = physeqG %>%
  aggregate_taxa(level = "Genus") %>%
  microViz::tax_sort(by="name") %>%
  microbiome::transform(transform = "compositional") %>%
  plot_composition(.,average_by = "sugar_cat") + 
  guides(fill = guide_legend(ncol = 1)) + 
  labs(x = "Added Sugar Tertile", y = "Relative abundance") +
  scale_fill_manual("Genus", values = mycolors) +
  scale_x_discrete(breaks=c("0","1","2"),labels=c("Low", "Intermediate", "High")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.title.y=element_text(size=20,face="bold"),
        axis.text.x=element_text(size=18,angle = -45, hjust=0),axis.text.y=element_text(size=18,color="black"),
        legend.title = element_text(size=18,color="black",face="bold"), legend.text = element_text(size=18,color="black")) 
```

## Differential abundance of genera

### Diet as continuous variable
```{r, warning=FALSE, message=FALSE}
hei_maslin = MaAsLin2.plus(ps = physeqGenus,
                            output="HEI",
                            fixed_effects = c("age","sex","race","platform","hei_iqr"),
                            reference = c("sex,1","race,0","platform,0"))

fiber_maslin = MaAsLin2.plus(ps = physeqGenus,
                            output="Fiber",
                            fixed_effects = c("age","sex","race","platform","fiber"),
                            reference = c("sex,1","race,0","platform,0"))

sugar_maslin = MaAsLin2.plus(ps = physeqGenus,
                            output="Sugar",
                            fixed_effects = c("age","sex","race","platform","sugar"),
                            reference = c("sex,1","race,0","platform,0"))
```

### Diet as categorical variable for trend test
```{r, warning=FALSE, message=FALSE}
hei_maslin2 = MaAsLin2.plus(ps = physeqGenus,
                            output="HEI",
                            fixed_effects = c("age","sex","race","platform","hei_cat2"),
                            reference = c("sex,1","race,0","platform,0"))

fiber_maslin2 = MaAsLin2.plus(ps = physeqGenus,
                            output="Fiber",
                            fixed_effects = c("age","sex","race","platform","fiber_cat2"),
                            reference = c("sex,1","race,0","platform,0"))


sugar_maslin2 = MaAsLin2.plus(ps = physeqGenus,
                            output="Sugar",
                            fixed_effects = c("age","sex","race","platform","sugar_cat2"),
                            reference = c("sex,1","race,0","platform,0"))


hei_taxadata = data.frame(cbind(hei_maslin2%>%select(Kingdom:Genus,coef:qval,fc:fc_up),
                                hei_maslin%>%select(coef:qval,fc:fc_up)))
fiber_taxadata = data.frame(cbind(fiber_maslin2%>%select(Kingdom:Genus,coef:qval,fc:fc_up),
                                  fiber_maslin%>%select(coef:qval,fc:fc_up)))
sugar_taxadata = data.frame(cbind(sugar_maslin2%>%select(Kingdom:Genus,coef:qval,fc:fc_up),
                                  sugar_maslin%>%select(coef:qval,fc:fc_up)))


hei_taxa = hei_taxadata %>%
  filter(qval<0.05&qval.1<0.05) %>%
  mutate(Genus=ifelse(Genus=="UC5-1-2E3","UC5-1-2E3 (Lachnospiraceae family genus)",Genus)) %>%
  ggplot(aes(y=reorder(Genus,-fc,sum), x=fc, fill=Phylum)) +
  scale_y_discrete(limits = rev,labels=function(x) stringr::str_wrap(gsub("_", " ", x, fixed=TRUE), width = 30))+
  geom_bar(stat="identity", width=0.5,color="black") +
  geom_errorbar(aes(xmin=fc_ll, xmax=fc_up), width=.2,position=position_dodge(.9)) +
  labs(y = "Genus", x = expression(bold(Log[2]~"Fold Change")), title = "HEI Tertile") +
  geom_vline(xintercept=0) +
  xlim(c(-5,5)) +
  scale_fill_manual("Phylum", values = c("#A6CEE3","#1F78B4","#33A02C")) + 
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.text.x=element_text(size=18, hjust=0),
        axis.title.y=element_text(size=20,face="bold"),axis.text.y=element_text(size=18),plot.title = element_text(size=20,face="bold"),
        legend.title = element_text(size=18,face="bold"), legend.text = element_text(size=18)) 

fiber_taxa = fiber_taxadata %>%
  filter(qval<0.05&qval.1<0.05) %>%
  mutate(Genus=ifelse(Genus=="UC5-1-2E3","UC5-1-2E3 (Lachnospiraceae family genus)",Genus)) %>%
  ggplot(aes(y=reorder(Genus,-fc,sum), x=fc, fill=Phylum)) +
  scale_y_discrete(limits = rev,labels=function(x) stringr::str_wrap(gsub("_", " ", x, fixed=TRUE), width = 30))+
  geom_bar(stat="identity", width=0.5,color="black") +
  geom_errorbar(aes(xmin=fc_ll, xmax=fc_up), width=.2,position=position_dodge(.9)) +
  labs(y = "Genus", x = expression(bold(Log[2]~"Fold Change")), title = "Dietary Fiber Tertile") +
  geom_vline(xintercept=0) +
  ylim(c(-4,4)) +
  scale_fill_manual("Phylum", values = c("#A6CEE3","#1F78B4","#33A02C")) +  
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.text.x=element_text(size=18, hjust=0),
        axis.title.y=element_text(size=20,face="bold"),axis.text.y=element_text(size=18),plot.title = element_text(size=20,face="bold"),
        legend.title = element_text(size=18,face="bold"), legend.text = element_text(size=18)) 

sugar_taxa = sugar_taxadata %>%
  filter(qval<0.05&qval.1<0.05) %>%
  ggplot(aes(y=reorder(Genus,-fc,sum), x=fc, fill=Phylum)) +
  scale_y_discrete(limits = rev,labels=function(x) stringr::str_wrap(gsub("_", " ", x, fixed=TRUE), width = 30))+
  geom_bar(stat="identity", width=0.5,color="black") +
  geom_errorbar(aes(xmin=fc_ll, xmax=fc_up), width=.2,position=position_dodge(.9)) +
  labs(y = "Genus", x = expression(bold(Log[2]~"Fold Change")), title = "Added Sugar Tertile") +
  geom_vline(xintercept=0) +
  scale_fill_manual("Phylum", values = c("#33A02C","#FB9A99")) +  
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.text.x=element_text(size=18, hjust=0),
        axis.title.y=element_text(size=20,face="bold"),axis.text.y=element_text(size=18),plot.title = element_text(size=20,face="bold"),
        legend.title = element_text(size=18,face="bold"), legend.text = element_text(size=18)) 
```

# Functional pathway analyses

## Filtering

```{r}
#Filter
any(taxa_sums(py) == 0)
py1 <- prune_taxa(taxa_sums(py) > 0, py)
ntaxa(py)
ntaxa(py1)

#Generate a prevalence table (number of samples each taxa occurs in) for each taxa.
prevdf = apply(X = otu_table(py1),
               MARGIN = 2,
               FUN = function(x){sum(x > 0)})

prevdf = data.frame(Prevalence = prevdf,
                    TotalAbundance = taxa_sums(py1),
                    tax_table(py1))

prevdf$pathway = rownames(prevdf)
pathway2Filter = prevdf$pathway[prevdf$TotalAbundance<100]

py3 = subset_taxa(py1, !rownames(tax_table(py1)) %in% pathway2Filter)
py3

prevalenceThreshold = 0.20 * nsamples(py3)
prevalenceThreshold

# Execute prevalence filter, using `prune_taxa()` function
prevdf1 = subset(prevdf, ta1 %in% get_taxa_unique(py3, taxonomic.rank = "ta1"))

keepTaxa = rownames(prevdf1)[(prevdf1$Prevalence >= prevalenceThreshold)]
py4 = prune_taxa(keepTaxa, py3)
py4
```

## Alpha diversity analysis

```{r}
shannon <- vegan::diversity(py4@otu_table, index = "shannon")
alphadiv <- cbind(py4@sam_data, shannon)

x1=summary(lm(shannon~hei_cat+age+sex+race+platform,data=alphadiv))$coefficients
x1.1=summary(lm(shannon~hei_cat2+age+sex+race+platform,data=alphadiv))$coefficients
x1.2=summary(lm(shannon~hei_iqr+age+sex+race+platform,data=alphadiv))$coefficients

x2=summary(lm(shannon~fiber_cat+age+sex+race+platform,data=alphadiv))$coefficients
x2.1=summary(lm(shannon~fiber_cat2+age+sex+race+platform,data=alphadiv))$coefficients
x2.2=summary(lm(shannon~fiber+age+sex+race+platform,data=alphadiv))$coefficients

x3=summary(lm(shannon~sugar_cat+age+sex+race+platform,data=alphadiv))$coefficients
x3.1=summary(lm(shannon~sugar_cat2+age+sex+race+platform,data=alphadiv))$coefficients
x3.2=summary(lm(shannon~sugar+age+sex+race+platform,data=alphadiv))$coefficients

table2=rbind(alpha_func(x1,x1.1,x1.2),alpha_func(x2,x2.1,x2.2),alpha_func(x3,x3.1,x3.2))
```

## Principal Component Analysis

```{r, warning=FALSE, message=FALSE}
data_otu_filt_rar = data.frame(otu_table(py4)) # create a separated file
dist_bc = as.matrix(vegdist(data_otu_filt_rar, method = "bray")) 

# a peek at the first five rows / columns
dist_bc[1:5, 1:5]

# calculate PCOA using Phyloseq package
pcoa_bc = ordinate(py4, "PCoA", "bray") 
mds_dat<-data.frame(pcoa_bc$vectors)
mds_dat$PEGID=row.names(mds_dat)
mds2<-merge(mds_dat,data.frame(py4@sam_data),by="PEGID")

mody <- capscale(pcoa_bc$vectors~ 1)
eig=eigenvals(mody)
head(eig / sum(eig))

clr1 =head(eig / sum(eig))[1]
clr2 =head(eig / sum(eig))[2]

set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+hei_cat2,
       data=d, permutations=9999, method="bray")$aov.tab
set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+fiber_cat2,
       data=d, permutations=9999, method="bray")$aov.tab
set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+sugar_cat2,
       data=d, permutations=9999, method="bray")$aov.tab

set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+hei_iqr,
       data=d, permutations=9999, method="bray")$aov.tab
set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+fiber,
       data=d, permutations=9999, method="bray")$aov.tab
set.seed(123)
adonis(data_otu_filt_rar~age+sex+race+platform+sugar,
       data=d, permutations=9999, method="bray")$aov.tab
```

## Differential abundance of pathways

### Diet as continuous variable
```{r, warning=FALSE, message=FALSE}
hei_maslin3 = MaAsLin2.plus(ps = py4,
                            output="HEI",
                            fixed_effects = c("age","sex","race","platform","hei_iqr"),
                            reference = c("sex,1","race,0","platform,0"))

fiber_maslin3 = MaAsLin2.plus(ps = py4,
                            output="Fiber",
                            fixed_effects = c("age","sex","race","platform","fiber"),
                            reference = c("sex,1","race,0","platform,0"))
```

### Diet as categorical variable for trend test
```{r, warning=FALSE, message=FALSE}
hei_maslin4 = MaAsLin2.plus(ps = py4,
                            output="HEI",
                            fixed_effects = c("age","sex","race","platform","hei_cat2"),
                            reference = c("sex,1","race,0","platform,0"))

fiber_maslin4 = MaAsLin2.plus(ps = py4,
                            output="Fiber",
                            fixed_effects = c("age","sex","race","platform","fiber_cat2"),
                            reference = c("sex,1","race,0","platform,0"))


hei_pathdata = data.frame(cbind(hei_maslin4%>%select(ta1:ta2,coef:qval,fc:fc_up),
                                hei_maslin3%>%select(coef:qval,fc:fc_up)))
fiber_pathdata = data.frame(cbind(fiber_maslin4%>%select(ta1:ta2,coef:qval,fc:fc_up),
                                  fiber_maslin3%>%select(coef:qval,fc:fc_up)))

hei_path = hei_pathdata %>%
  filter(qval<0.05&qval.1<0.05) %>%
  mutate(ta2 = Hmisc::capitalize(ta2),
         Superclass = factor(c("Degradation/Utilization/Assimilation","Degradation/Utilization/Assimilation","Biosynthesis","Biosynthesis"),
                             levels = c("Biosynthesis","Degradation/Utilization/Assimilation"))) %>%
  ggplot(aes(y=reorder(ta2,-fc,sum), x=fc, fill=Superclass)) +
  scale_y_discrete(limits = rev, labels = function(x) stringr::str_wrap(x, width = 35))+
  geom_bar(stat="identity", width=0.5,color="black") +
  geom_errorbar(aes(xmin=fc_ll, xmax=fc_up), width=.2,
                 position=position_dodge(.9)) +
  labs(y = "MetaCyc Pathway", x = expression(bold(Log[2]~"Fold Change")), title = "HEI Tertile") +
  geom_vline(xintercept=0) +
  scale_fill_manual("Superclass Function",values = c("#BC80BD","#FB9A99")) + 
  xlim(c(-2,2)) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.text.x=element_text(size=18, hjust=1),
        axis.title.y=element_text(size=20,face="bold"),axis.text.y=element_text(size=18),
        legend.title = element_text(size=18,face="bold"),legend.text = element_text(size=18),
        plot.title = element_text(size=20,face="bold")) 

fiber_path = data.frame(fiber_pathdata) %>%
  filter(qval<0.05&qval.1<0.05) %>%
  mutate(ta2 = Hmisc::capitalize(ta2),
         Superclass = factor(c("Degradation/Utilization/Assimilation","Biosynthesis","Biosynthesis","Biosynthesis"),
                             levels = c("Biosynthesis","Degradation/Utilization/Assimilation"))) %>%
  ggplot(aes(y=reorder(ta2,-fc,sum), x=fc, fill=Superclass)) +
  scale_y_discrete(limits = rev, labels = function(x) stringr::str_wrap(x, width = 35))+
  geom_bar(stat="identity", width=0.5,color="black") +
  geom_errorbar(aes(xmin=fc_ll, xmax=fc_up), width=.2,
                 position=position_dodge(.9)) +
  labs(y = "MetaCyc Pathway", x = expression(bold(Log[2]~"Fold Change")), title = "Dietary Fiber Tertile") +
  geom_vline(xintercept=0) +
  scale_fill_manual("Superclass Function",values = c("#BC80BD","#FB9A99")) + 
  xlim(c(-4,1)) +
  theme_bw() +
  theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.background = element_blank(),
        axis.title.x=element_text(size=20,face="bold"),axis.text.x=element_text(size=18, hjust=1),
        axis.title.y=element_text(size=20,face="bold"),axis.text.y=element_text(size=18),
        legend.title = element_text(size=18,face="bold"),legend.text = element_text(size=18),
        plot.title = element_text(size=20,face="bold")) 
```
